tests/Microsoft.Identity.Test.Integration.netcore/Infrastructure/SeleniumExtensions.cs

```diff
@@ -24,35 +24,54 @@ public static class SeleniumExtensions
 
         public static IWebDriver CreateDefaultWebDriver()
         {
-            ChromeOptions options = new ChromeOptions();
-            ChromeDriver driver;
-
-            // ~2x faster, no visual rendering
-            // remove when debugging to see the UI automation
-            options.AddArguments("headless");
-
-            var env = Environment.GetEnvironmentVariable("CHROMEWEBDRIVER");
-            if (string.IsNullOrEmpty(env))
+            // ---------- Chrome launch flags ----------
+            var options = new ChromeOptions();
+            options.AddArguments(
+                "--headless=new",          // modern headless mode (remove this for debugging)
+                "--disable-gpu",
+                "--window-size=1920,1080",
+                "--remote-allow-origins=*",
+                "--disable-dev-shm-usage"); // avoids crashes in low-memory containers
+
+            // ---------- Pick a driver binary ----------
+            // 1) Prefer explicit env-var so devs can override locally
+            var driverDir = Environment.GetEnvironmentVariable("CHROMEWEBDRIVER");
+
+            // 2) Otherwise use the folder where CI drops a matching chromedriver
+            if (string.IsNullOrEmpty(driverDir))
             {
-                driver = new ChromeDriver(options);
-            }
-            else
-            {
-                driver = new ChromeDriver(env, options);
+                driverDir = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "drivers");
```

Comment:
Also why is this needed? Is this something standard on the build agents?

Reply:
this is more of a fallback mechanism to for local testing or if we use this in CI